server/functions/setval.go

```diff
@@ -54,11 +57,38 @@ var setval_text_int64_boolean = framework.Function3{
 			return nil, err
 		}
 		// TODO: this should take a regclass as the parameter to determine the schema
-		schema, err := core.GetCurrentSchema(ctx)
+		schema, relation, err := parseRelationName(ctx, val1.(string))
 		if err != nil {
 			return nil, err
 		}
-
-		return val2.(int64), collection.SetVal(schema, val1.(string), val2.(int64), val3.(bool))
+		return val2.(int64), collection.SetVal(schema, relation, val2.(int64), val3.(bool))
 	},
 }
+
+// parseRelationName parses the schema and relation name from a relation name string, including trimming any
+// identifier quotes used in the name. For example, passing in 'public."MyTable"' would return 'public' and 'MyTable'.
+func parseRelationName(ctx *sql.Context, name string) (schema string, relation string, err error) {
```

Comment:
This seems fine but it's going to fail for table / schema names that contain a `.`. Longer term we need to resolve the reference at creation time and quote it properly before storing it as a default value.